src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -804,5 +816,75 @@ public void Dispose()
                     _operation._currentWriter.Write("]}}");
             }
         }
+
+        public class AttachmentsBulkInsert : IDisposable
```

Comment:
You don't need IDisposable here. And there should be only one instance of it (like with counters).